tests/test_base_client.py

```diff
@@ -0,0 +1,52 @@
+import unittest
+import recurly
+from recurly import Resource
+from .mock_resources import MyResource, MySubResource
+from .mock_client import MockClient
+import unittest.mock as mock
+from unittest.mock import Mock, MagicMock
+
+
+def get_resource_client(success):
+    conn = MagicMock()
+    conn.request = MagicMock(return_value=None)
+    response = MagicMock()
+    if success:
+        response.status = 200
+        response.read.return_value = bytes('{"object": "my_resource"}', "UTF-8")
+    else:
+        response.status = 404
+        response.read.return_value = bytes(
+            """
+            {
+                "error": {
+                   "type": "not_found",
+                   "message": "Couldn't find Resource with id = 123",
+                   "params": [{"param": "resource_id"}]
+               }
+            }
+            """,
+            "UTF-8",
+        )
+
+    conn.getresponse = MagicMock(return_value=response)
+
+    return mock.patch("http.client.HTTPSConnection", return_value=conn)
+
+
+class TestBaseClient(unittest.TestCase):
+    def test_api_version(self):
+        client = MockClient("subdomain", "apikey")
+        self.assertEqual(client.api_version(), "v2018-08-09")
+
+    def test_successful_GET_200(self):
+        with get_resource_client(True) as conn:
+            client = MockClient("subdomain", "apikey")
+            resource = client.get_resource("123")
+            self.assertEqual(type(resource), MyResource)
+
+    def test_failure_GET_404(self):
+        with get_resource_client(False) as conn:
```

Comment:
Why do you have `as conn` here (and in the other test)? We're not using the value here and if you remove it, the tests still pass.

I'd actually love to have a longer, general conversation around this pattern you chose.

Reply:
This will be used to assert the incoming call is correct. Example: `conn.request.assert_called_with("GET", "/resources/123", None)`